### PR TITLE
Introduce slot description and new dbus properties

### PR DESCRIPTION
**Pull request by ejoerns**
Also transforms former signal 'ProgressUpdated' to property to allow reading its whenever required.

### CODE REVIEW

File: src/config_file.c [hunk 1]
```diff
@@ -154,6 +154,10 @@ gboolean load_config(const gchar *filename, RaucConfig **config, GError **error)
 			slot->name = g_intern_string(value);
 			g_free(value);
 
+			slot->description = g_intern_string(g_key_file_get_string(key_file, groups[i], "description", NULL));
```
Comment (jluebbe) [inline src/config_file.c]: This should not be an interned string. It will need freeing in free_slot().

File: src/service.c [hunk 2]
```diff
@@ -161,6 +225,8 @@ static void r_on_name_acquired(GDBusConnection *connection,
 	if (r_context()->config->autoinstall_path)
 		auto_install(r_context()->config->autoinstall_path);
 
+	set_slot_status();
+
```
Comment (jluebbe) [inline src/service.c]: This should be done after loading the config, so that the slot information is always available.

File: src/service.c [hunk 3]
```diff
@@ -110,6 +111,69 @@ static gboolean auto_install(const gchar *source) {
...
+	slot_status_tuples = g_new(GVariant*, slot_number);
+
+	res = determine_slot_states(NULL);
+	if (!res)
+		return FALSE;
+
+	g_hash_table_iter_init(&iter, r_context()->config->slots);
+	while (g_hash_table_iter_next(&iter, &key, &value)) {
+		gchar *name = key;
```
Comment (jluebbe) [inline src/service.c]: Create a per slot function from the body of this loop.

File: src/service.c [hunk 4]
```diff
@@ -110,6 +111,69 @@ static gboolean auto_install(const gchar *source) {
...
+
+			}
+			r_umount_slot(slot, NULL);
+			g_clear_pointer(&slotstatuspath, g_free);
+		}
+
+		if (slot->state == ST_BOOTED)
+			booted = slot;
+
+		slot_status = g_new(GVariant*, 3);
```
Comment (jluebbe) [inline src/service.c]: Make the dbus-property update a separate function.

**Comment by ejoerns** [comment]
Replaced by #57

**Pull request closed without merging**